### DISCUSSION TITLE
Copying behavior changes depending on file type and size

### DISCUSSION THREAD

**Discussion by nponsard** [dokan-dev/dokany#1204] [Q&A]
Hello,

I am writing a custom file system driver in rust with dokan-rust. I have the latest version of Dokany installed on my computer.

When I try to copy a file to this drive, the behavior changes depending on the file type and size.

- pdf : windows explorer displays an error File too large
- jpg : Copying files over ~500kB works, with smaller files windows explorer displays a file too large error
- exe : works with any size

I noticed that the function called changes. When it will end up in an error, it doesn't call `set_file_attributes` after the write but directly calls `delete_file`.

My guess would be that I don't do the right thing in `write_file` or don't return the right informations in `get_file_information`. 
But I can't seem to find what.

Has anyone experienced this ?

Thanks

**Comment by nponsard** [comment]
Oh I found the difference between files that work and files that don't, it's not related to type or size.

It doesn't work when the original file has this in the properties 
![image](https://github.com/dokan-dev/dokany/assets/56651588/6de1c64f-e2ef-4285-bc75-c03bac140a00)

**Comment by Olof-Lagerkvist** [comment]
That attribute is stored as an alternate data stream to the file. Could it be that your `GetVolumeInformation` sets a flag indicating that you support named alternate data streams, but then have not implemented them correctly? As far as I know, when copying such files to file systems that do not support alternate data streams at all, such as FAT32, Windows simply ignores this information and just copies the main data stream of the file.

**Comment by nponsard** [comment]
Sorry for not updating this discussion. I found what I did wrong, I did not set the mount flag `ALT_STREAM` when mounting my file system. I thought the flag present meant that the filesystem supports alternate streams but apparently it's the inverse.
